From 34c01c17468a7f55ffb3bbbf73e8d83c49246812 Mon Sep 17 00:00:00 2001
From: Guillaume Papin <guillaume.papin@epitech.eu>
Date: Tue, 11 Dec 2018 00:01:21 +0100
Subject: [PATCH 2/3] CMake: use Threads::Threads instead of
 CMAKE_THREAD_LIBS_INIT

Using the more modern imported target does not require a more recent CMake.
---
 CMakeLists.txt                   | 2 +-
 cmake/cppgraphqlgen-config.cmake | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d3f670..a1b395b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ add_library(graphqlservice
   Introspection.cpp
   ${CMAKE_BINARY_DIR}/graphqlservice/IntrospectionSchema.cpp)
 target_link_libraries(graphqlservice PRIVATE taocpp::pegtl)
-target_link_libraries(graphqlservice PUBLIC ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(graphqlservice PUBLIC Threads::Threads)
 target_include_directories(graphqlservice
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
diff --git a/cmake/cppgraphqlgen-config.cmake b/cmake/cppgraphqlgen-config.cmake
index ac7ced6..e398243 100644
--- a/cmake/cppgraphqlgen-config.cmake
+++ b/cmake/cppgraphqlgen-config.cmake
@@ -15,4 +15,5 @@ The following import targets are created
 
 include(CMakeFindDependencyMacro)
 find_package(pegtl REQUIRED)
+find_package(Threads REQUIRED)
 include("${CMAKE_CURRENT_LIST_DIR}/cppgraphqlgen-targets.cmake")
-- 
2.10.0

